docs/designs/cli.md

```diff
@@ -0,0 +1,80 @@
+# ElasticDL Command-line Client Tool
+
+## Background
+
+ElasticDL is a Kubernetes-Native deep learning framework. As it runs
+distributed training/prediction/evaluation jobs in a cluster, we need a client
+to submit the job to the cluster. Currently we have one but it's tightly
+coupled with the main package. User need pip install the whole elasticdl
+package and lots of dependencies and it's too heavy.
+
+The main functionality of the client is *building image for ElasticDL job* and
+*submitting ElasticDL job*. It only has depedency on docker and Kubernetes Api.
+In this doc, we are focusing on a light-weight client CLI of ElasticDL.
+
+## User Story
+
+1. User develops model and the directory structure is as follows:
+
+    ```TEXT
+    a_directory
+    - wide_and_deep.py
+    requirements.txt
+    ```
+
+2. Generate an Dockerfile.  
+
+    Input the command:
+
+    ```bash
+    cd ${model_root_path}
+    elasticdl zoo init [base_image_name]
+    ```
+
+    `base_image_name` is optional and the default value is `python:3`. The generated Dockerfile example is:
```

Comment:
Why is the default value is `python:3` not ubuntu or other images?

Reply:
Python 3 is the fundamental dependency for ElasticDL and TensorFlow 2.x. 
I just tried ubuntu:18.04 and don't find python in it.

So I choose the docker image `python` which provides the fundamental execution environment.